plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java

```diff
@@ -43,19 +48,29 @@
         public static final Setting<KeyStoreType> KEYSTORE_TYPE_SETTING =
             new Setting<>("cloud.azure.management.keystore.type", KeyStoreType.pkcs12.name(), KeyStoreType::fromString,
                 Property.NodeScope, Property.Filtered);
+
+        public static final Setting<URI> ENDPOINT_SETTING = new Setting<URI>("cloud.azure.management.endpoint",
```

Comment:
I wonder if we want to document this? I'm perfectly fine without adding any doc regarding this as IMO no one but tests should modify this.
That being said if someone ask for all cluster settings with `GET _cluster/settings?include_defaults`, this one will popup.